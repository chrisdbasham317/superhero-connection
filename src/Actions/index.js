export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
});

export const hasErrored = msg => ({
  type: 'HAS_ERRORED',
  msg
});

export const setHeroes = heroes => ({
  type: 'SET_HEROES',
  heroes
})