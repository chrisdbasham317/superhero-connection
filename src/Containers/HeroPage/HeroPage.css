.section--hero-page,
.div--loaded-hero-data-render {
  background-image: url('https://i.pinimg.com/originals/2f/69/14/2f69142f096557e7b5a4efed8a4ca66f.jpg');
  display: flex;
  color: #ffffff;
  flex-direction: row;
}

.section--name-pic {
  display: flex;
  flex-direction: column;
  margin: 2vw;
}

.h1--hero-name {
  font-size: 2rem;
}

.link--hero-battle-btn {
  align-items: center;
  align-self: center;
  background: #ffffff;
  border-radius: 20px;
  display: flex;
  height: 5vh;
  justify-content: center;
  margin: 5vh;
  width: 60%;
}

.link--add-hero {
  color: #000000;
  text-decoration: unset;
}

.section--hero-info {
  background: #000000;
  border: 3px solid grey;
  display: flex;
  flex-direction: column;
  font-size: 1.2rem;
  opacity: .7;
  padding-left: 3vw;
  height: fit-content;
  margin: 16vh 0 5vh 0;
  width: 57vw;
}

.h2--info-label {
  font-size: 1.65rem;
  margin: 1vh 0;
  text-decoration: overline;
}

.li--stats {
  text-transform: uppercase;
}

p {
  font-size: 1.25rem;
  text-transform: capitalize;
}