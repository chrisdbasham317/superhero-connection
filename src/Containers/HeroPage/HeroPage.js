import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import loadingIcon from '../../images/loading-icon.jpg';


const HeroPage = ({ id, heroes }) => {
  const heroId = parseInt(id.split('').filter(elem => elem !== '}').join(''));
  const currentHero = heroes.filter(hero => hero.id === heroId)[0];
  const attributes = currentHero ? Object.keys(currentHero.powerstats) : null;
  const aliases = currentHero ? currentHero.biography.aliases.map(alias => <li>{alias}</li>) : null;
  const powerStats = currentHero ? attributes.map(stat => <li>{`${stat}: ${currentHero.powerstats[stat]}`}</li>) : null;
  return (
    <section className='section--hero-page'>
      {!currentHero && loadingIcon}
      {currentHero && 
      <div className='div--loaded-hero-data-render'>
        <h1 className='h1--hero-name'>{currentHero.name}</h1>
        <img src={currentHero.images.md} alt={`${currentHero.name} photo`} />
        <ul className='ul--alias-list'>
          <p>Aliases</p>
          {aliases}
        </ul>
        <ul className='ul--stats-list'>
          <p>Attributes</p>
          {powerStats}
        </ul>
        <p>Birth Place: {currentHero.biography.placeOfBirth}</p>
        <p>First Appearance: {currentHero.biography.firstAppearance}</p>
        <p>Publisher: {currentHero.biography.publisher}</p>
        <p>Alignment: {currentHero.biography.alignment}</p>
        <div className='link--hero-battle-btn'>
          <Link to='/battle'>Choose Hero</Link>
        </div>
      </div>    
      }
    </section>
  )
}

export const mapStateToProps = ({ heroes }) => ({
  heroes
})

export default connect(mapStateToProps)(HeroPage);