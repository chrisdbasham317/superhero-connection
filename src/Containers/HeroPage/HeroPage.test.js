import React from 'react';
import { shallow } from 'enzyme';
import { setCombatant1, setCombatant2 } from '../../Actions';
import { HeroPage, mapStateToProps, mapDispatchToProps } from './HeroPage';


describe('HeroPage', () => {
  let wrapper
  const mockHeroes = [
    {
      id: 4,
      name: 'superman',
      biography: {
        aliases: ['Man of Steel']
      },
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 100,
      }
    },
    {
      id: 5,
      name: 'superboy',
      biography: {
        aliases: ['Conner']
      },
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 85,
      }
    }
  ]  
  let mockSetCombatant1 = jest.fn();
  let mockSetCombatant2 = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<HeroPage
      id={'4}'}
      heroes={mockHeroes}
      combatant1={'superman'}
      setCombatant1={mockSetCombatant1}
      setCombatant2={mockSetCombatant2}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setCombatant2 when combatant1 is defined', () => {
    wrapper.find('.link--add-hero').simulate('click');
    expect(mockSetCombatant2).toHaveBeenCalledWith('superman');
  });

  it('should return an object with the heroes array, and combatant1', () => {
    const mockState = {
      heroes: mockHeroes,
      combatant1: 'superman'      
    }
    const expected = {
      heroes: mockHeroes,
      combatant1: 'superman'
    };
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });  
})