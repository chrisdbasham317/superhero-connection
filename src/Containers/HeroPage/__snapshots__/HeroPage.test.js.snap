// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroPage should match the snapshot 1`] = `
<section
  className="section--hero-page"
>
  <div
    className="div--loaded-hero-data-render"
  >
    <section
      className="section--name-pic"
    >
      <h1
        className="h1--hero-name"
      >
        superman
      </h1>
      <img
        alt="superman"
        src="medium image"
      />
      <div
        className="link--hero-battle-btn"
      >
        <Link
          className="link--add-hero"
          onClick={[Function]}
          to="/battle"
        >
          Choose Hero
        </Link>
      </div>
    </section>
    <section
      className="section--hero-info"
    >
      <h2
        className="h2--info-label"
      >
        Aliases
      </h2>
      <ul
        className="ul--alias-list"
      >
        <li
          key="Man of Steel"
        >
          Man of Steel
        </li>
      </ul>
      <h2
        className="h2--info-label"
      >
        Attributes
      </h2>
      <ul
        className="ul--stats-list"
      >
        <li
          className="li--stats"
          key="strength"
        >
          strength: 100
        </li>
        <li
          className="li--stats"
          key="intelligence"
        >
          intelligence: 100
        </li>
      </ul>
      <h2
        className="h2--info-label"
      >
        Birth Place:
      </h2>
      <p />
      <h2
        className="h2--info-label"
      >
        First Appearance:
      </h2>
      <p />
      <h2
        className="h2--info-label"
      >
        Publisher:
      </h2>
      <p />
      <h2
        className="h2--info-label"
      >
        Alignment:
      </h2>
      <p />
    </section>
  </div>
</section>
`;
